### DISCUSSION TITLE
Retrieving the Complete Change History of a Container or SharedTree

### DISCUSSION THREAD

**Discussion by Yishin26** [microsoft/FluidFramework#21239] [Q&A] [answered]
Hi,

I am currently working with the Fluid Framework and need assistance with retrieving the complete change history of a container or a sharedTree. Could you please provide detailed instructions or point me to the relevant documentation that explains how to access and view the entire change history?

Thank you for your help 🥹

**Comment by nmsimons** [comment, marked as answer]
There is currently no built in way to get the full change history for a Fluid document. We are currently investigating the best ways to surface such information so I am very interested in your scenarios.

**Reply by Yishin26** [reply]
We are looking for a collaborative software solution that can provide users with a detailed CRUD operation history, including functionalities like replaying actions. It would be ideal if this could be achieved without relying on external storage methods.  
Thank you 🥹

**Reply by nmsimons** [reply]
Thanks for the additional context. Do you want this information persisted across sessions or just for the current session?

**Reply by Yishin26** [reply]
Either is fine, leave it up to you to decide :-)
